pkg/local_object_storage/peapod/peapod.go

```diff
@@ -0,0 +1,144 @@
+package peapod
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"sync"
+	"time"
+
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"go.etcd.io/bbolt"
+)
+
+type batch struct {
+	initErr error
+
+	tx *bbolt.Tx
+
+	commitErr   error
+	chCommitted chan struct{}
+
+	bktRootMtx sync.Mutex
+	bktRoot    *bbolt.Bucket
+}
+
+// Peapod provides storage for relatively small NeoFS binary object (peas).
+// Peapod is a single low-level key/value database optimized to work with big
+// number of stored units.
+type Peapod struct {
+	bolt *bbolt.DB
+
+	currentBatchMtx sync.RWMutex
+	currentBatch    *batch
+
+	chClose     chan struct{}
+	chFlushDone chan struct{}
+}
+
+var rootBucket = []byte("root")
+
+// New returns initialized Peapod instance located at the given path. Resulting Peapod
+// must be finally closed.
+func New(path string) (*Peapod, error) {
+	db, err := bbolt.Open(path, 0600, &bbolt.Options{
+		Timeout: 100 * time.Millisecond, // to handle flock
+	})
+	if err != nil {
+		return nil, fmt.Errorf("open BoltDB instance: %w", err)
+	}
+
+	err = db.Update(func(tx *bbolt.Tx) error {
+		_, err = tx.CreateBucketIfNotExists(rootBucket)
+		return err
+	})
+	if err != nil {
+		return nil, fmt.Errorf("create root bucket in BoltDB instance: %w", err)
+	}
+
+	res := &Peapod{
+		bolt:        db,
+		chClose:     make(chan struct{}),
+		chFlushDone: make(chan struct{}),
+	}
+
+	res.beginNewBatch()
+
+	go res.flushLoop()
+
+	return res, nil
+}
+
+// Close syncs data and closes the database.
+func (x *Peapod) Close() error {
+	close(x.chClose)
+	<-x.chFlushDone
+	return x.bolt.Close()
+}
+
+func (x *Peapod) flushLoop() {
+	defer close(x.chFlushDone)
+
+	ticker := time.NewTicker(10 * time.Millisecond)
```

Comment:
Do we plan to adjust the value from the external config? I think even no, this is a `magic const` here, it would be better to put it in the `const` section above

Reply:
right now it doesn't make sense to expose this interval as config because we don't have strict model of how it changes the performance (i reached the current value experimentally). Having dedicated const is not bad but it'll be used in single place, i don't mind